What's new for .NET 9 Preview 6 (#41745)
dotnet/docs: docs/core/tools/buildcheck-rules/bc0102.md

---
title: "BC0102: Double writes"
description: Learn about .NET SDK error BC0102, which occurs when multiple tasks attempt to write to a single file.
ms.topic: error-reference
ms.date: 07/10/2024
f1_keywords:
- BC0102
---
# BC0102

**This article applies to:** ✔️ .NET 9 SDK and later versions

> This rule flags a similar problem as [BC0101 - Shared output path](bc0101.md), but with higher granularity. It's not recommended that multiple tasks attempt to write to a single file. Such behavior can lead to nondeterminism of a build (results can be dependent on the order of task execution if they belong to independent projects) or to lost updates.

Reviewer: > higher granularity

Could be use a more clear term?

> If you want multiple tasks to update a single file in a one-by-one pipeline fashion, give each intermediate output a distinct name. Distinct files prevent silent mixups if any of the tasks in the chain are skipped or removed.

Reviewer: ```suggestion
If multiple tasks need to update a single file in a one-by-one pipeline fashion, each intermediate output should be given a distinct name. Distinct files prevent silent mixups if any of the tasks in the chain are skipped or removed.
```
